workorder-recheck.lua

```diff
@@ -1,20 +1,97 @@
 -- Resets the selected work order to the `Checking` state
---[====[
-
-workorder-recheck
-=================
-Sets the status to ``Checking`` (from ``Active``) of the selected work order (in the ``j-m`` or ``u-m`` screens).
-This makes the manager reevaluate its conditions.
-]====]
-local scr = dfhack.gui.getCurViewscreen()
-if df.viewscreen_jobmanagementst:is_instance(scr) then
-    local orders = df.global.world.manager_orders
-    local idx = scr.sel_idx
-    if idx < #orders then
-        orders[idx].status.active = false
+
+--@ module = true
+
+local widgets = require('gui.widgets')
+local overlay = require('plugins.overlay')
+
+local function set_current_inactive()
+    local scrConditions = df.global.game.main_interface.info.work_orders.conditions
+    if scrConditions.open then
+        local order = scrConditions.wq
+        order.status.active = false
     else
-        qerror("Invalid order selected")
+        qerror("Order conditions is not open")
     end
-else
-    qerror('Must be called on the manager screen (j-m or u-m)')
 end
+
+-- -------------------
+-- RecheckOverlay
+--
+
+local focusString = 'dwarfmode/Info/WORK_ORDERS/Conditions'
+
+RecheckOverlay = defclass(RecheckOverlay, overlay.OverlayWidget)
+RecheckOverlay.ATTRS{
+    default_pos={x=6,y=8},
+    default_enabled=true,
+    viewscreens=focusString,
+    frame={w=19, h=3},
+    -- frame_style=gui.MEDIUM_FRAME,
+    -- frame_background=gui.CLEAR_PEN,
+}
+
+local function areTabsInTwoRows()
+    -- get the tile above the order status icon
+    local pen = dfhack.screen.readTile(7, 7, false)
+    -- in graphics mode, `0` when one row, something else when two (`67` aka 'C' from "Creatures")
+    -- in ASCII mode, `32` aka ' ' when one row, something else when two (`196` aka '-' from tab frame's top)
+    return (pen.ch ~= 0 and pen.ch ~= 32)
+end
+
+function RecheckOverlay:updateTextButtonFrame()
+    local twoRows = areTabsInTwoRows()
+    if (self._twoRows == twoRows) then return false end
+
+    self._twoRows = twoRows
+    local frame = twoRows
+            and {b=0, l=0, r=0, h=1}
+            or  {t=0, l=0, r=0, h=1}
+    self.subviews.button.frame = frame
+
+    return true
+end
+
+function RecheckOverlay:init()
+    self:addviews{
+        widgets.TextButton{
+            view_id = 'button',
+            -- frame={t=0, l=0, r=0, h=1}, -- is set in `updateTextButtonFrame()`
+            label='recheck',
+            key='CUSTOM_CTRL_A',
+            on_activate=set_current_inactive,
```

Comment:
This button should probably be disabled if the workorder is already inactive. e.g. add:
```lua
enabled=is_current_active,
```